### PR TITLE
Fix FTP dependency tracking

### PR DESCRIPTION
**Pull request by doncem**
As inspected in production - there is no FTP tracking happening automagically for us. After attempts to apply signature as MS suggests no luck achieved. Going back to old way of just manually tracking the dependency to keep track of this activity

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: lib/AI-Agent.xml [hunk 1]
```diff
@@ -18,11 +18,6 @@
         <Class name="uk.gov.hmcts.reform.sendletter.services.encryption.PgpEncryptionUtil">
```
Comment (banderous) [inline lib/AI-Agent.xml]: Are any of these being tracked automatically or can we get rid of this file?

File: src/main/java/uk/gov/hmcts/reform/sendletter/services/ftp/FtpClient.java [hunk 2]
```diff
@@ -80,19 +95,31 @@ public void upload(LocalSourceFile file, boolean isSmokeTestFile) {
                         }
                     })
                     .collect(toList());
+
+                insights.trackFtpReportDownload(Duration.between(now, Instant.now()), true);
```
Comment (banderous) [inline src/main/java/uk/gov/hmcts/reform/sendletter/services/ftp/FtpClient.java]: you could combine these last two insights tracking calls by using a finally block after the exception handler, here and in a couple of other places

File: lib/AI-Agent.xml (hunk 1 above)
Comment (doncem) [reply lib/AI-Agent.xml]: Encryption and pdf creator tracks - no problem. for some reason a mixture of void methods are not working. Next would be task scheduler + silencing info logs. Only need to know single hit of the task and it's success rate

File: src/main/java/uk/gov/hmcts/reform/sendletter/services/ftp/FtpClient.java [hunk 3]
```diff
@@ -52,9 +63,13 @@ public void upload(LocalSourceFile file, boolean isSmokeTestFile) {
                 String path = String.join("/", folder, file.getName());
                 sftp.getFileTransfer().upload(file, path);
 
+                success = true;
```
Comment (banderous) [inline src/main/java/uk/gov/hmcts/reform/sendletter/services/ftp/FtpClient.java]: I hadn't spotted that we tracked success! Happy with either approach.

Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/sendletter/services/ftp/FtpClient.java]: yea, only two key features of dependency: success and time took

File: src/main/java/uk/gov/hmcts/reform/sendletter/services/ftp/FtpClient.java [hunk 4]
```diff
@@ -32,18 +35,26 @@
...
         this.configProperties = configProperties;
+        this.insights = insights;
     }
     // endregion
 
     public void upload(LocalSourceFile file, boolean isSmokeTestFile) {
+        Instant now = Instant.now();
+
         runWith(sftp -> {
+            boolean success = false;
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/sendletter/services/ftp/FtpClient.java]: can we rename this to isSuccess please ?

**Pull request merged**